graphics/blender/Portfile

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+
+name                blender
+version             2.82a
+revision            0
+categories          graphics multimedia
+platforms           darwin
+license             GPL-2+
+maintainers         @jasonliu--
+
+homepage            https://www.blender.org
+description         3-D computer graphics creation suite
+long_description    Blender is a ${description}. It contains a set of \
+                    tools that can be used for all phases of the 3-D \
+                    computer graphics pipeline, including modeling, \
+                    sculpting, texturing, rigging, animation, \
+                    simulation, rendering, compositing, motion \
+                    tracking, and video editing. It can be used for \
+                    creating animated films, visual effects, art, 3-D \
+                    printing, motion graphics, interactive 3-D \
+                    applications, and computer games.
+
+master_sites        https://download.blender.org/source/
+use_xz              yes
+checksums           rmd160  477b18c4d51ed6e5214158f1438c2c9a4317b2e7 \
+                    sha256  4de00030201d14e98cce3d3ed61b3c72267fb07dbd46ba779dc261a3d56beba3 \
+                    size    39890216
+
+compiler.cxx_standard 2011
+# We need to blacklist Xcode Clang/LLVM, because Blender needs to
+# use tools such as llvm-config, which are not included in Xcode.
+# (llvm-config is needed because Blender links to libLLVM. libLLVM is
+# also not included in Xcode.)
+# (Reference: https://stackoverflow.com/questions/32959627/where-is-the-system-llvm-config-on-macos)
+compiler.blacklist clang
+# Even though we need to blacklist Xcode Clang/LLVM, we still need to
+# make the rest of Xcode available to the build system. If we don't set
+# use_xcode to yes, then the build will fail, because Blender's CMake
+# scripts still need to use other parts of Xcode (e.g., the xcodebuild
+# command) besides just Xcode Clang.
+use_xcode yes
+cmake.install_prefix ${applications_dir}
+
+depends_lib-append  port:python38 \
+                    port:openal-soft \
+                    port:alembic \
+                    port:opensubdiv \
+                    port:jack \
+                    port:libsndfile \
+                    port:fftw-3 \
+                    port:zlib \
+                    port:freetype \
+                    port:openexr \
+                    port:ffmpeg \
```

Comment:
Can ffmpeg-devel be used instead? If so, we normally write this dependency as:

`path:bin/ffmpeg:ffmpeg
`

Reply:
I'm not sure that using a `-devel` package would be a good idea. Blender usually uses version of third-party dependencies that are slightly older than their current versions. Trying to use a version that is even newer than the current version might break things.

Also, grepping through the portfiles, it looks like it's split 50/50 between writing the dependency as `port:ffmpeg` and `path:bin/ffmpeg:ffmpeg`. Is there some sort of advantage to specifically making the dependency based on the binary executable, as opposed to the entire ffmpeg port? Blender never actually uses the ffmpeg binary. It does, however, use various ffmpeg libraries, such as libavcodec and libavformat.

Reply:
Using a path dependency is currently the only way to allow a conditional dependency like this. It doesn't actually matter which file is used in the path. It's a nice allowance for people who want to have ffmpeg-devel installed because it conflicts with the ffmpeg port.

Reply:
So, does that mean that using `path:bin/ffmpeg:ffmpeg` would allow the blender port to install successfully regardless of whether the user has either `ffmpeg-devel` or `ffmpeg` installed?

Reply:
I don't know, that's why I'm asking you. You need to test that blender builds and works when ffmpeg-devel is installed in place of ffmpeg.

Reply:
I just tried it, and using `path:bin/ffmpeg:ffmpeg` with `ffmpeg-devel` installed doesn't work. It does cause the build to fail a bit later: using `port:ffmpeg`, the build fails before the fetch phase even starts; using `path:bin/ffmpeg:ffmpeg`, the build will manage to get all the way to the configure phase before the build fails. However, in both cases, MacPorts will detect that the `ffmpeg` port isn't installed and active, and the build will abort.

Reply:
That does not make sense. Adding either `path:bin/ffmpeg:ffmpeg` or `port:ffmpeg` to the `depends_lib` will ensure that they are installed and active first before building blender.

Reply:
On my machine, adding either of those to the `depends_lib` does indeed ensure that ffmpeg is installed and active, but if ffmpeg-devel is already installed and active beforehand, the MacPorts build will complain that ffmpeg is not installed, and _cannot_ be installed due to conflicting with ffmpeg-devel. So it aborts.

Reply:
Exactly, they conflict. That's why we want this path dependency. To test, you have to manually deactivate ffmpeg and then install ffmpeg-devel.

Reply:
Yes, that's exactly what I did to do the testing.